src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -120,32 +127,48 @@ public EventBridgeWriter(EventBridgeAsyncClient ebClient, EventBridgeSinkConfig
    * @param records
    * @return list of all records with additional status information
    */
-  public List<EventBridgeWriterRecord> putItems(List<EventBridgeWriterRecord> records) {
+  public List<Result<EventBridgeWriterErrorSinkRecord, Pair<PutEventsRequestEntry, SinkRecord>>>
```

Comment:
The information we need in the `EventBridgeSinkTask` are the failed `SinkRecord`(s) w/ the Error (JSON serialization or batch size). Do we need the (detailed) information about the successful send `SinkRecord`(s)? @maschnetwork?

Reply:
Not yet but do you see any disadvantages of returning the full list of success and error and let the Task filter?

Reply:
No, I would only return the failures with there `SinkRecord`(s). Maybe the number of successful send as convenience.

Reply:
If we agree that there are no disadvantages I would go with returning all records as we currently do with the new result class. Task can then just filter for failed and we stay flexible.